Sources/TuistCache/ContentHashing/GraphContentHasher.swift

```diff
@@ -1,79 +1,97 @@
+        self.init(
+            contentHasher: contentHasher,
+            sourceFilesContentHasher: SourceFilesContentHasher(contentHasher: contentHasher),
+            targetActionsContentHasher: TargetActionsContentHasher(contentHasher: contentHasher),
+            coreDataModelsContentHasher: CoreDataModelsContentHasher(contentHasher: contentHasher)
+        )
     }
 
-    public func contentHashes(for graph: Graph) throws -> [TargetNode: String] {
-        var visitedNodes: [TargetNode: Bool] = [:]
+    public init(
+        contentHasher: ContentHashing,
+        sourceFilesContentHasher: SourceFilesContentHashing,
+        targetActionsContentHasher: TargetActionsContentHashing,
+        coreDataModelsContentHasher: CoreDataModelsContentHashing
+    ) {
+        self.contentHasher = contentHasher
+        self.sourceFilesContentHasher = sourceFilesContentHasher
+        self.coreDataModelsContentHasher = coreDataModelsContentHasher
+        self.targetActionsContentHasher = targetActionsContentHasher
+    }
+
+    // MARK: - GraphContentHashing
 
+    public func contentHashes(for graph: TuistCore.Graph) throws -> [TargetNode: String] {
+        var visitedNodes: [TargetNode: Bool] = [:]
         let hashableTargets = graph.targets.values.flatMap { (targets: [TargetNode]) -> [TargetNode] in
             targets.compactMap { target in
                 if self.isCacheable(target, visited: &visitedNodes) { return target }
                 return nil
             }
         }
-        let hashes = try hashableTargets.map { try makeContentHash(of: $0) }
+        let hashes = try hashableTargets.map { try hash(targetNode: $0) }
         return Dictionary(uniqueKeysWithValues: zip(hashableTargets, hashes))
     }
 
+    // MARK: - Private
+
     fileprivate func isCacheable(_ target: TargetNode, visited: inout [TargetNode: Bool]) -> Bool {
         if let visitedValue = visited[target] { return visitedValue }
-
         let isFramework = target.target.product == .framework
         let noXCTestDependency = target.sdkDependencies.first(where: { $0.name == "XCTest.framework" }) == nil
         let allTargetDependenciesAreHasheable = target.targetDependencies.allSatisfy { isCacheable($0, visited: &visited) }
-
         let cacheable = isFramework && noXCTestDependency && allTargetDependenciesAreHasheable
         visited[target] = cacheable
         return cacheable
     }
 
-    fileprivate func makeContentHash(of targetNode: TargetNode) throws -> String {
-        // TODO: extend function to consider build settings, compiler flags, dependencies, and a lot more
-        let sourcesHash = try hashSources(of: targetNode)
-        let productHash = try hash(string: targetNode.target.productName)
-        let platformHash = try hash(string: targetNode.target.platform.rawValue)
-        return try hash(strings: [sourcesHash, productHash, platformHash])
-    }
+    private func hash(targetNode: TargetNode) throws -> String {
```

Comment:
To be consistent with what you have done with other models, what about extracting this into a `TargetContentHasher`?

Reply:
👍  done